examples/s4u/actor/s4u_actor.cpp

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2006-2015. The SimGrid Team. All rights reserved.          */
+
+/* This program is free software; you can redistribute it and/or modify it
+ * under the terms of the license (GNU LGPL) which comes with this package. */
+
+#include <xbt/sysdep.h>
+#include <mutex>
+
+#include "simgrid/s4u.h"
+
+#define NB_ACTOR 2
+
+XBT_LOG_NEW_DEFAULT_CATEGORY(s4u_test, "a sample log category");
+
+// simgrid::s4u::Mutex mtx; //FIXME generate error -> You must run MSG_init before using MSG
+
+//Create an actor as a c++ functor
+class Worker {
+simgrid::s4u::Mutex *pMtx;
+int *pResults;
+public:
+  Worker(int  *res, simgrid::s4u::Mutex *mtx) : pMtx(mtx),  pResults(res) {}; 
+  // Define the code of the actor
+  void operator()() {
+    // Do the calculation
+    simgrid::s4u::this_actor::execute(1000);    
+    
+    // lock the mutex before enter in th critical section
+    pMtx->lock();
+    XBT_INFO("Hello s4u, I'm ready to compute");
+    
+    // And finaly add it to the results
+    *pResults += 1;
+    XBT_INFO("I'm done, good bye");
+    
+    //Then unlock the mutex, so other threads will be able to update their results
+    pMtx->unlock();
+  }
+};
+
+// This class is an example of how to use lock_guard with simgrid mutex
+class WorkerLockGuard {
+simgrid::s4u::Mutex *pMtx;
+int *pResults;
+public:
+  WorkerLockGuard(int  *res, simgrid::s4u::Mutex *mtx) : pMtx(mtx),  pResults(res) {}; 
+  void operator()() {
+    
+    simgrid::s4u::this_actor::execute(1000);    
+    
+    // Simply use the std::lock_guard like this
+    std::lock_guard<simgrid::s4u::Mutex> lock(*pMtx);
+    
+    // then you are in a safe zone
+    XBT_INFO("Hello s4u, I'm ready to compute");
+    // update the results
+    *pResults += 1;
+    XBT_INFO("I'm done, good bye");
+  }
+};
+
+class MainActor {
+public:
+  void operator()() {
+    int *res = new int;
+    simgrid::s4u::Mutex *mtx = new simgrid::s4u::Mutex();
+    *res = 0;
+    simgrid::s4u::Actor* workers[NB_ACTOR*2];
```

Comment:
I get this warning:

    simgrid/examples/s4u/actor/s4u_actor.cpp:68:26: error: variable ‘workers’ set but not used [-Werror=unused-but-set-variable]